tests/test_install_mender_sh.py

```diff
@@ -245,6 +261,123 @@ def test_upgrade_all_packages(
         check_installed(generic_debian_container, "mender-configure")
 
 
+@pytest.mark.usefixtures("script_server")
+class TestUpgradeMenderV4:
+    def test_upgrade_from_v3_to_v4_to_build(
+        self, generic_debian_container,
+    ):
+        # Install Mender 3.7.0 software with mender-client 3.5.2
+        local_apt_repo_from_upstream_packages(
+            generic_debian_container,
+            [
+                "m/mender-client/mender-client_3.5.2-1+debian+buster_amd64.deb",
+                "m/mender-connect/mender-connect_2.2.0-1+debian+buster_amd64.deb",
+                "m/mender-configure/mender-configure_1.1.1-1+debian+buster_all.deb",
+            ],
+            "/release_3_7_0",
```

Comment:
Not against it, but in the spirit of future splitting, maybe I would omit 3.7 references here, and only use 3.x and 4.x client references instead.

Same with this other one, and with the comments.